### PR TITLE
chore: move @[simp] from pred_le to sub_one_le

### PR DESCRIPTION
**Pull request by kim-em**
(We already have a simp lemma unfolding `pred` to `· - 1`.)

### CODE REVIEW

**Comment by semorrison** [comment]
@nomeata, could you assist me on this one? `tests/lean/guessLexTricky.lean` appears to be the only thing failing, and I can't work out what is going wrong.

File: src/Init/Data/Nat/Basic.lean [hunk 1]
```diff
@@ -278,10 +278,13 @@ theorem succ_sub_succ_eq_sub (n m : Nat) : succ n - succ m = n - m := by
   | zero      => exact rfl
   | succ m ih => apply congrArg pred ih
 
-@[simp] theorem pred_le : ∀ (n : Nat), pred n ≤ n
+theorem pred_le : ∀ (n : Nat), pred n ≤ n
   | zero   => Nat.le.refl
   | succ _ => le_succ _
 
+@[simp] theorem sub_one_le (n : Nat) : n - 1 ≤ n :=
+  pred_le n
```
Comment (nomeata) [inline src/Init/Data/Nat/Basic.lean:286]: Isn't this lemma subsumed by `Nat.sub_le`, which could just be made `@[simp]`?

**Comment by nomeata** [comment]
Pushed a work-around

File: tests/lean/guessLexTricky.lean [hunk 2]
```diff
@@ -39,32 +39,32 @@ macro_rules | `(tactic|search_lex $ts:tacticSeq) => `(tactic| (
...
+-- termination_by (y, 0, 1)
 decreasing_by
   simp_wf
-  search_lex solve
-    | decreasing_trivial
-    | apply Nat.bitwise_rec_lemma; assumption
+  try -- the need for `try` here is fishy
+      -- the proof with explicit `termination_by` does not need it, so it should not throw
+      -- GuessLex off, but without `try` it does
+      -- This appeared after #4522, which made Nat.sub_one_le a simp lemma
```
Comment (semorrison) [inline tests/lean/guessLexTricky.lean:57]:
```suggestion
      -- This appeared after #4522, which made Nat.sub_le a simp lemma
```

**Pull request merged**